lib/typhoeus/response/status.rb

```diff
@@ -81,6 +92,16 @@ def first_header_line
           end
         end
       end
+
+      # :nodoc:
+      def has_good_response_code?
+        response_code >= 200 && response_code < 300
```

Comment:
To simplify, I'd say that anything `>=200 && < 400` is successful here. You're right in the sense that people can check and handle codes manually if they need to.